### ISSUE TITLE
Possible feature request: Track My Pipe

### ISSUE THREAD

**Issue by ale5000-git** [TeamNewPipe/NewPipe#2278]
An idea just come to my mind.
Beside personal preference, it could be useful if someone created an app called for "Track My Pipe" (completely separated from NewPipe) implemented similarly to the Google Contacts Sync Adapter but that periodically sync the NewPipe subscription and other things to the Google/YouTube account for the ones that need it.

**Comment by chopin2712** [comment]
Yes ! I want to use New Pipe for replacement to YouTube but I don't have my 200 subscriptions to add one by one. Great Idea !

**Comment by Stypox** [comment]
@chopin2712 you can export your subscriptions from yt and import them in NewPipe

**Comment by ale5000-git** [comment]
@Stypox: Yes but the key here is continual automatic sync.

**Comment by nv95** [comment]
Do anyone know how to fetch user's subscriptions without proprietary api and interaction with user? I think, it is not possible (

**Comment by gkeegan** [comment]
It might be possible with Python's Requests module.

**Comment by ale5000-git** [comment]
@nv95: The code could parse the web interface (without API) like NewPipe.

For the account it could use [microG](https://github.com/microg) (it replace Google Play Service for those that don't know).

**Comment by gkeegan** [comment]
I don't think NewPipe will use microG, even considering that it is open source. It still adds that tie to Google than many people do not want.

From the readme of this repository: "you don't need a YouTube account to use NewPipe". Signing in to a Google account has been shot down many times before here, and won't ever be allowed.

The Requests module would be better, as it could sign in to YouTube like you would in a web browser, which could be done in the background. Then it could download the HTML of the subscriptions page, and parse it for channel links. 

That is the best solution to syncing that I think NewPipe would allow, and I'm not even sure this would be allowed.

@ale5000-git
@theScrabi : how would you feel about this? It could maybe be provided in a separate fork, like the legacy fork is doing. It would need some maintainers though.

**Comment by ale5000-git** [comment]
@gkeegan: If you read the first message I was talking about a separate apk (that use microG).
That separate apk should be able to send subscription to NewPipe but it should be still separate.

**Comment by ale5000-git** [comment]
Since an image is better than 1000 words: ![Track My Pipe](https://user-images.githubusercontent.com/15793015/57249427-2e19ae00-7045-11e9-9b05-a33ed6248147.png)

**PS:** It is just an example, nothing currently exist.

**Comment by gkeegan** [comment]
Ohhh, sorry I see what you mean now. Forgive me. This would definitely be possible to be done.

I could probably whip something up on Python with the module I said in a few days once I get back to my PC. Someone else would have to put it into an app and get the automatic syncing working though.

**Comment by Gymcap** [comment]
Maybe a version that syncs to invidious?

**Comment by woj-tek** [comment]
@Gymcap - that would actually be awesome, with synchronisation of watch history (i.e. https://invidio.us/feed/history)
